### DISCUSSION TITLE
Avatar: WCAG issue - Images must have alternate text

### DISCUSSION THREAD

**Discussion by eXpertise7** [primefaces/community#642] [PrimeNG] [answered]
Hi,

I believe there are accessibility issues with `p-avatar` component on `primeng@16.9.0` (stable PrimeNG version 16).

![image](https://github.com/primefaces/community/assets/17203747/167e18f9-b103-40b7-90f8-a8ce4d3cfa6c)

![image](https://github.com/primefaces/community/assets/17203747/2f1bd02d-60b1-44d6-9ba4-b9f863d86d4d)

Setting up `role="img"` with/or `ariaLabel `or `ariaLabelledBy` (and even: `aria-label` and `aria-labelled-by`) at `p-avatar` component declaration at template as suggested on PrimeNG `p-avatar` documentation **does not help** with this issue.

![image](https://github.com/primefaces/community/assets/17203747/32fb7a1f-4ee7-43f3-acf0-feb85d01bc9e)

By the way, shouldn't aria attributes fill the nested `img` element, not the it's parent `div` element?

Also, one question, how's that that `p-avatar` does not support `alt` `@Input` for accessibility, that's main thing for image accessibility.

Thanks in advance.

Kind regards,
Robert Medved

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for the report, we've fixed the issue and the fix will be available on next release https://github.com/primefaces/primeng/issues/14296
